src/parser/mod.rs

```diff
@@ -745,6 +745,8 @@ pub fn parse(input: &str) -> TeraResult<Vec<Node>> {
                     Rule::boolean => "`true` or `false`".to_string(),
                     Rule::ident => "an identifier".to_string(),
                     Rule::dotted_ident => "a dotted identifier (identifiers separated by `.`)".to_string(),
+                    Rule::dotted_square_bracket_ident => "an identifier (identifiers separated by `.` or `[]`s)".to_string(),
+                    Rule::square_brackets => "an identifier, string or integer inside `[]`s".to_string(),
```

Comment:
looks like there is some overlap on the error messages which can cause some confusing results like a test below.

Maybe modify the `Rule::ident` message to be more precise and mention it can only be a-z etc?